mod.ts

```diff
@@ -6,3 +6,4 @@ export * from "./src/handle_callback.ts";
 export * from "./src/get_session_id.ts";
 export * from "./src/sign_in.ts";
 export * from "./src/sign_out.ts";
+export * from "./src/fresh_plugin.ts";
```

Comment:
This worries me, pulling framework specific stuff into a fairly general module.

Maybe framework specific modules could be separate from the mod.ts...

```ts
import { kvOAuthPlugin } from "https://deno.land/x/deno_kv_oauth@$VERSION/fresh_plugin.ts";
```

Reply:
I think it's fine. Are you able to elaborate further?

Reply:
`fresh_plugin.ts` imports from `"$fresh/server.ts"`, so if I import `mod.ts` from this project, it will (at least for typechecking) require `$fresh` to be aliased in your import map.

I tried adding this in a module, to import `mod.ts` from this specific branch:

```ts
import { signIn } from "https://ghuc.cc/denoland/deno_kv_oauth@fresh-plugin/mod.ts";
```

If I run `deno check` on this it complains:

```
error: Relative import path "$fresh/server.ts" not prefixed with / or ./ or ../ and not in import map from "https://raw.githubusercontent.com/denoland/deno_kv_oauth/fresh-plugin/src/fresh_plugin.ts"
```

This is a pain if you aren't using Fresh (not that I have anything against Fresh).

Personally I find `mod.ts` to be an anti-pattern, dragging us back to days of the proverbial npm/OO dependency jungle (_I just wanted the banana, but instead I got the monkey holding the banana, and the entire jungle_).

So I consume deno_kv_oauth by aliasing `.../src/` and just import the individual modules I need (eg: `import type { signIn } from "$deno_kv_oauth/sign_in.ts"`), and I can continue doing so. But I imagine many users, esp newer users, will simply import `mod.ts`, and if they aren't using Fresh, they are going to hit this problem, which could be discouraging, which would be a shame.

Reply:
Actually, I'd go so far as to suggest that maybe this plugin should be in Fresh itself, alongside the twind plugin,
so that importing it follows the same pattern, for example:

```ts
import twindPlugin from "$fresh/plugins/twind.ts";
import authPlugin from "$fresh/plugins/deno_kv_oauth.ts";
```

Reply:
https://ghuc.cc/ is a redirection service for raw github content, check it out, it's very handy.

I'm not trying to import a fresh plugin, the `@fresh-plugin` part is just the name of this branch in the git repo, I can see how that looks confusing.

I'm importing the general `mod.ts` for deno_kv_oauth, but in doing so, it transiently attempts to import the fresh plugin, which I don't want imported, which is my whole point.

Reply:
> I'd go so far as to suggest that maybe this plugin should be in Fresh itself

valid point.  
is there actually any criteria if and when a plugin is added to the main repository?

Reply:
> This is a pain if you aren't using Fresh (not that I have anything against Fresh).

Ok, agreed.

> Actually, I'd go so far as to suggest that maybe this plugin should be in Fresh itself, alongside the twind plugin

Hm... I'd say I'm 50/50 on that idea. Any thoughts, @marvinhagemeister and @lucacasonato?